api.js

```diff
@@ -195,7 +195,7 @@ function handlePaths(files) {
 		files = [
 			'test.js',
 			'test-*.js',
-			'test/*.js'
```

Comment:
@sindresorhus: Well, I _may_ have oversold it a little, but here's my setup:

I start with a file structure like this:

[//]: # "├──, └──"
[//]: # "These comments look way out of line, but they're not."

    my-app
    ├── src/                     # Code
    |   ├── foo.js               # Some fascinating module
    |   ├── foo.test.unit.js     # The unit (doesn't exit the process, <150ms/test) tests for `foo.js`
    |   ├── bar.js               # Another fascinating module
    |   ├── bar.test.int.js      # The focused integration tests for `bar.js`
    |   └── ...                  # Other stuff too
    ├── test/                    # Tests that don't belong to a single source file, utils, mocks, fixtures, etc.
    |   ├── utils/               # Test utilities
    |   |   ├── findPort.js      # Find an available port.
    |   |   ├── portAvail.js     # Test if a port is available.
    |   |   └── ...              # Other stuff too
    |   ├── fixture/             # Test fixtures
    |   |   ├── some-test        # Find an available port.
    |   |   └── ...              # Other stuff too
    |   ├── mocks/               # Some mocks 
    |   |   ├── SomeClass.js     # A mock for `SomeClass`
    |   |   └── ...              # Other stuff too
    |   ├── webapp.test.e2e.js   # End2End tests for the web app
    |   ├── mobile.test.e2e.js   # End2End tests for the mobile app
    |   ├── api.test.e2e.js      # End2End tests for the API
    |   ├── setup.js             # Register some global helper functions, and bluebird.
    |   └── ...                  # Other stuff too
    └── ...                      # Other stuff too

Then you use [`app-module-path`](npm.im/app-module-path) for `my-app`, and then your tests look like this:

```javascript
import test from 'ava';
import rewire from 'rewire';
import findPort from 'test/utils/findPort';
import portAvail from 'test/utils/portAvail';
import FakeSomeClass from 'test/mock/SomeClass';

test.beforeEach(async t => {
    t.context.port = await findPort();
    t.context.foo = rewire('./foo');
});

test(async t => {
    await t.context.foo.createServer(t.context.port);
    t.ok(await portAvail(t.context.port));
});
```

And the actual files:

```javascript
import http from 'http';
import config from 'app/config';
import connectToDb from 'app/db';

export async function createServer(port) {
    var server = http.createServer();
    var db = await connectToDb();
    await Promise.promisify(::server.listen)(port);
    return { server, db };
};
```

And I have build scripts set up so I can test, test:cov or watch based on test type:

```bash
$ npm run test
> npm run test:all

    100 tests passing.

$ npm run test:cov
> npm run test:all:cov
> ava
    100 tests passing.
    
    100% coverage.

$ npm run test:{unit,int,e2e}
> ...

$ npm run test:{unit,int,e2e}:cov
> ...

$ npm run watch:{unit,int,es2}
> ...
Watching...

```